packages/client/hmi-client/src/components/responsive-matrix/matrix.fs.glsl.ts

```diff
@@ -0,0 +1,30 @@
+export default `
+#version 300 es
+
+precision lowp usampler2D;
+precision highp sampler2D;
+precision highp float;
+
+in vec2 vUvs;
+
+uniform sampler2D uColor;
+uniform usampler2D uMicroRow;
+uniform usampler2D uMicroCol;
+
+out vec4 fragColor;
+
+vec4 getColorFromRowCol(sampler2D tex, int index) {
+	int texWidth = textureSize(tex, 0).x;
+	int col = index % texWidth;
+	int row = index / texWidth;
+	return texelFetch(tex, ivec2(col, row), 0);
+}
+
+void main() {
+	float microColLen = float(textureSize(uMicroCol, 0).x);
+	float microRowLen = float(textureSize(uMicroRow, 0).x);
+	uint colIndex = texelFetch(uMicroCol, ivec2(int(vUvs.x * microColLen), 0), 0).r;
+	uint rowIndex = texelFetch(uMicroRow, ivec2(int(vUvs.y * microRowLen), 0), 0).r;
+
+	fragColor = texelFetch(uColor, ivec2(colIndex, rowIndex), 0);
```

Comment:
It feels like you are using sampler2D as if it is a sampler1D - is sampler1D not supported in webGL?

Also it seems like `getColorFromRowCol` is not being used - is that in your later PRs or can we remove?

Reply:
A bit of comment describing what the main method would be good to have.

Reply:
sampler1d would be ideal for this implementation but in order to get around the texture size limit (around 8192) which is much too low a limit for microrow/col, i planned to convert it to using 2d texture.

`getColorFromRowCol` is a function i stole from grafer that should help with this, but again, not implemented yet.

i should probably do a commit implementing 2d tex